File: src/main/java/com/pedigree/ui/ToolbarFactory.java
package com.pedigree.ui;

import javafx.scene.control.Button;
import javafx.scene.control.ToolBar;

public class ToolbarFactory {

    private final Runnable onZoomIn;
    private final Runnable onZoomOut;

    public ToolbarFactory(Runnable onZoomIn, Runnable onZoomOut) {
        this.onZoomIn = onZoomIn;
        this.onZoomOut = onZoomOut;
    }

    public ToolBar create() {
        Button bZoomIn = button("Zoom +", onZoomIn);
        Button bZoomOut = button("Zoom -", onZoomOut);
        return new ToolBar(bZoomIn, bZoomOut);
    }

    private static Button button(String text, Runnable action) {
        Button b = new Button(text);
        b.setOnAction(e -> { if (action != null) action.run(); });
        return b;
    }
}

File: src/main/kotlin/com/family/tree/ui/ToolbarFactory.kt
package com.family.tree.ui

import javafx.scene.control.Button
import javafx.scene.control.ToolBar

class ToolbarFactory(private val onZoomIn: Runnable?, private val onZoomOut: Runnable?) {

    fun create(): ToolBar {
        val bZoomIn = button("Zoom +", onZoomIn)
        val bZoomOut = button("Zoom -", onZoomOut)
        return ToolBar(bZoomIn, bZoomOut)
    }

    private fun button(text: String, action: Runnable?): Button {
        val b = Button(text)
        b.setOnAction { action?.run() }
        return b
    }
}
